packages/react-day-picker/src/contexts/SelectRange/SelectRangeContext.tsx

```diff
@@ -118,63 +118,45 @@ export function SelectRangeProviderInternal({
-      if (selected.to) {
-        modifiers.range_middle = [
-          {
-            after: selected.from,
-            before: selected.to
-          }
-        ];
-        if (max || min) {
-          modifiers.disabled = [
-            (date: Date) => {
-              if (
-                max &&
-                selected.to &&
-                selected.from &&
-                isBefore(date, selected.from)
-              ) {
-                const diff = differenceInCalendarDays(selected.to, date);
-                if (diff >= max) {
-                  return true;
-                }
+      const to = selected.to ? selected.to : selected.from;
+      modifiers.range_middle = [
+        {
+          after: selected.from,
+          before: to
+        }
+      ];
+      if (max || min) {
+        modifiers.disabled = [
+          (date: Date) => {
+            if (max && to && selected.from && isBefore(date, selected.from)) {
+              const diff = differenceInCalendarDays(to, date);
+              if (diff >= max) {
+                return true;
               }
-              if (
-                max &&
-                selected.to &&
-                selected.from &&
-                isAfter(date, selected.to)
-              ) {
-                const diff = differenceInCalendarDays(date, selected.from);
-                if (diff >= max) {
-                  return true;
-                }
+            }
+            if (max && to && selected.from && isAfter(date, to)) {
+              const diff = differenceInCalendarDays(date, selected.from);
+              if (diff >= max) {
+                return true;
               }
-              if (min && selected.from && isBefore(date, selected.from)) {
-                const diff = differenceInCalendarDays(selected.from, date);
-                if (diff < min) {
-                  return true;
-                }
+            }
+            if (min && selected.from && isBefore(date, selected.from)) {
+              const diff = differenceInCalendarDays(selected.from, date);
+              if (diff < min) {
+                return true;
               }
-              if (
-                min &&
-                selected.to &&
-                selected.from &&
-                isAfter(date, selected.to)
-              ) {
-                const diff = differenceInCalendarDays(date, selected.from);
-                if (diff < min) {
-                  return true;
-                }
+            }
+            if (min && to && selected.from && isAfter(date, to)) {
+              const diff = differenceInCalendarDays(date, selected.from);
+              if (diff < min) {
+                return true;
               }
-              return false;
             }
-          ];
-        }
-        modifiers.range_end = [selected.to];
-      } else {
-        modifiers.range_end = [selected.from];
+            return false;
+          }
+        ];
       }
+      modifiers.range_end = [to];
```

Comment:
There are I think no tests for this code that you changed...care to write a few?